File: Device/DeviceController.Data/Solenoid.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DeviceController.Data
{
    public enum HardwareTypes { GPIO, Distributed, SPI};
    public class Solenoid
    {
        public int Id;
        public string Name;
        public string Description;
        public HardwareTypes HardwareType;
        public string Address;
        public int Value;
        public bool RequiresPump;
        public int DeviceId;
        public DateTime? CreatedAt;
        public DateTime? UpdatedAt;
    }
}

File: WebClient/src/app/model/solenoid.ts
export class ISolenoid {
    constructor(
        public Id: number,
        public Name: string,
        public Description: string,
        public HardwareType: string,
        public Address: string,
        public Value: number,
        public RequiresPump: boolean,
        public DeviceId: number
    ) {}
}
